uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/feature/SamlLoginWithLocalIdpIT.java

```diff
@@ -616,41 +727,48 @@ public void testCrossZoneSamlIntegration() throws Exception {
-        assertNotNull(beforeLogin);
-        assertNotNull(beforeLogin.getValue());
+            webDriver.get(baseUrl + "/logout.do");
+            webDriver.get(spZoneUrl + "/logout.do");
+            webDriver.get(idpZoneUrl+ "/logout.do");
+            webDriver.get(spZoneUrl + "/");
+            assertEquals(spZone.getName(), webDriver.getTitle());
+            Cookie beforeLogin = webDriver.manage().getCookieNamed("JSESSIONID");
+            assertNotNull(beforeLogin);
+            assertNotNull(beforeLogin.getValue());
+
+            List<WebElement> elements = webDriver
+                .findElements(By.xpath("//a[text()='" + samlIdentityProviderDefinition.getLinkText() + "']"));
+            assertNotNull(elements);
+            assertEquals(1, elements.size());
 
-        List<WebElement> elements = webDriver
-            .findElements(By.xpath("//a[text()='" + samlIdentityProviderDefinition.getLinkText() + "']"));
-        assertNotNull(elements);
-        assertEquals(1, elements.size());
+            WebElement element = elements.get(0);
+            assertNotNull(element);
 
-        WebElement element = elements.get(0);
-        assertNotNull(element);
-        element.click();
-        webDriver.findElement(By.xpath("//h1[contains(text(), 'Welcome to The Twiglet Zone[" + idpZoneId + "]!')]"));
-        webDriver.findElement(By.name("username")).clear();
-        webDriver.findElement(By.name("username")).sendKeys(idpZoneUserEmail);
-        webDriver.findElement(By.name("password")).sendKeys("secr3T");
-        webDriver.findElement(By.xpath("//input[@value='Sign in']")).click();
-        assertThat(webDriver.findElement(By.cssSelector("h1")).getText(), Matchers.containsString("Where to?"));
-        Cookie afterLogin = webDriver.manage().getCookieNamed("JSESSIONID");
-        assertNotNull(afterLogin);
-        assertNotNull(afterLogin.getValue());
-        assertNotEquals(beforeLogin.getValue(), afterLogin.getValue());
+            element.click();
+        try {
+            webDriver.findElement(By.xpath("//h1[contains(text(), 'Welcome to The Twiglet Zone[" + idpZoneId + "]!')]"));
+            webDriver.findElement(By.name("username")).clear();
+            webDriver.findElement(By.name("username")).sendKeys(idpZoneUserEmail);
+            webDriver.findElement(By.name("password")).sendKeys("secr3T");
+            webDriver.findElement(By.xpath("//input[@value='Sign in']")).click();
+            assertThat(webDriver.findElement(By.cssSelector("h1")).getText(), Matchers.containsString("Where to?"));
+            Cookie afterLogin = webDriver.manage().getCookieNamed("JSESSIONID");
+            assertNotNull(afterLogin);
+            assertNotNull(afterLogin.getValue());
+            assertNotEquals(beforeLogin.getValue(), afterLogin.getValue());
+        } catch (Exception e) {
+            Assert.
+            assertTrue("Http-Artifact binding is not supported",e instanceof NoSuchElementException);
```

Comment:
this assertion is wrong. there is no guarantee that NoSuchElementException means artifact binding not supported.